vmdb/app/models/ems_refresh/save_inventory.rb

```diff
@@ -173,6 +168,10 @@ def save_guest_devices_inventory(hardware, hashes)
         # Save the hardware to force an id if not found
         hardware.save! if hardware.id.nil?
         h[:network][:hardware_id] = hardware.id
+      else
+        # ensure save_network_inventory gets called
+        # it will be deleted
+        h[:network] = {}
```

Comment:
This seems wrong.  If a key doesn't exist, it is ignored (i.e. NOT deleted).  The hash builders in the parsers should make that conscious decision.  They can choose to not set the key (ignore), set the key with a nil value (delete or ensure does not exist), or set the key with a specific value (update or create).